simulator/mips/mips_instr.h

```diff
@@ -409,10 +409,16 @@ class BaseMIPSInstr
         {
             if ( index == 0)
                 v_src1 = value;
-            else
+            else if ( index == 1)
```

Comment:
What is the motivation for that check?

Reply:
Possible errors with index.

Reply:
That is called "defensive programming", as you do not trust your callers provide adequate inputs. That pattern has no advantages, as we guarantee that our inputs are always correct by unit test suites.

Reply:
OK, I'll remove it.